### DISCUSSION TITLE
Strain calculation: finite total strain

### DISCUSSION THREAD

**Discussion by LabrosV** [idaholab/moose#18036] [Q&A Modules: Solid mechanics] [answered]
Hi all,

According to this page (https://mooseframework.inl.gov/modules/tensor_mechanics/Strains.html), there is no finite total strain formulation in Moose. I want to solve a hyperelastic model with no plasticity and the incremental form (hypo-elastic) will cause inaccuracies for large strains.  I'm writing this post to discuss the changes in the code I have to make in order to include hyper-elasticity with a total strain formulation. 

Some initial thoughts/steps:
1. The properties _total_strain and _mechanical_strain will be defined in class that inherits from "ComputeStrainBase".
2. Modifying TensorMechanicsAction(Base) to call the above function if StrainAndIncrement::FiniteTotal.
3. The Finite Strain Hyperelastic stress will be calculated in a function which inherits from "ComputeStressBase".

After making these changes, I was able to solve some numerical examples but the jacobian was not correct (I used -snes_test_jacobian in a transient problem for comparison). I pushed forward my jacobian because the stress divergence kernel is expressed in the current configuration. I'm not sure if I have to make any changes in the solvers (nonlinear). 

I look forward to hearing your feedback/ideas about this topic.

Thanks,
Labros

**Comment by jiangwen84** [comment, marked as answer]
FYI, we have an active PR to add total finite strain formulation with correct Jacobian. see https://github.com/idaholab/moose/pull/17475

**Reply by dschwen** [reply]
Was just about to link to that, too. Feel free to chime in on that PR!
